Google.PowerShell/PubSub/GcpsMessage.cs

```diff
@@ -415,4 +415,132 @@ protected override void ProcessRecord()
+    /// and -AckId to provide a list of Ack Ids for that subscription. To send acknowledgement for messages object returned by
+    /// Get-GcpsMessage cmdlet, use the -InputObject parameter.
+    /// </para>
+    /// <example>
+    ///   <code>PS C:\> Send-GcpsAck -Subscription "my-subscription" -AckId "ackId"</code>
+    ///   <para>
+    ///   This command sends acknowledgement for message with Ack Id "ackId" from subscription "my-subscription" in the default project.
+    ///   </para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> Send-GcpsAck -Subscription "my-subscription" -AckId "ackId1", "ackId2" -Project "my-project"</code>
+    ///   <para>
+    ///   This command sends acknowledgement for messages with Ack Ids "ackId1" and "ackId2" from subscription"my-subscription"
+    ///   in the project "my-project".
+    ///   </para>
+    /// </example>
+    /// <example>
+    ///   <code>
+    ///   PS C:\> $messages = Get-GcpsMessage -Subscription "my-subscription"
+    ///   PS C:\> Send-GcpsAck -InputObject $messages
+    ///   </code>
+    ///   <para>
+    ///   This command sends acknowledgement for messages pulled from subscription "my-subscription"
+    ///   </para>
+    /// </example>
+    /// <para type="link" uri="(https://cloud.google.com/pubsub/docs/subscriber#receiving-pull-messages)">
+    /// [Receiving and Sending Acknowledge for Pull Messages]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommunications.Send, "GcpsAck")]
+    public class SendGcpsAck : GcpsCmdlet
+    {
+        private class ParameterSetNames
+        {
+            public const string ByObject = "ByObject";
+            public const string ByName = "ByName";
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// The project that the subscription belongs to. If not set via PowerShell parameter processing, will
+        /// default to the Cloud SDK's DefaultProject property.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByName)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The name of the subscription that the messages are pulled from.
+        /// This parameter is used with -AckId parameter.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 0, ParameterSetName = ParameterSetNames.ByName)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Subscription), Property =nameof(Subscription.Name))]
+        [Alias("Subscription")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The list of AckIds of the pulled messages from the provided subscription.
+        /// This parameter is used with -Name parameter.
```

Comment:
This is the only place you use "-Name" everywhere else you use "-Subscription".  It might be useful to either use Name in other places, mention it here or used Subscription.